### PR TITLE
gadgets/readmes: update all gadgets documentation

### PR DESCRIPTION
**Pull request by alban**
# gadgets/readmes: update all gadgets documentation

The templates are updated. When a new ebpf param is added in a gadget, the readmes can be updated **in place** with:
```
make -C gadgets readmes
```

4 gadgets didn't have a README.mdx (lsm, malloc, signal, ssl). This is now fixed.

Fixes: https://github.com/inspektor-gadget/inspektor-gadget/issues/3357

## How to use

```
make -C gadgets readmes
```

## Testing done

None

### CODE REVIEW

File: gadgets/README.template [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
-### `--{{ index $flag "key" }}`
-{{ index $flag "description" }}
-
-Default value: "{{ index $flag "defaultValue" }}"
-{{ end -}}
-{{- else -}}
-No flags.
-{{- end -}}
+
+TODO
```
Comment (mauriciovasquezbernal) [inline gadgets/README.template:34]: Can't it be generated too?

File: gadgets/trace_bind/gadget.yaml [hunk 2]
```diff
@@ -56,9 +56,9 @@ datasources:
           columns.width: 10
 ebpfParams:
   ignore_errors:
-    key: ignore_errors
+    key: ignore-errors
     alias: i
-    defaultValue: ""
+    defaultValue: "true"
```
Comment (mauriciovasquezbernal) [inline gadgets/trace_bind/gadget.yaml:61]: note: It's actually not needed anymore as we're getting these from the ebpf object. (I'll open an issue / PR to clean it up).

File: gadgets/snapshot_socket/README.mdx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 title: snapshot_socket
-sidebar_position: 35
+sidebar_position: 40
```
Comment (mauriciovasquezbernal) [inline gadgets/snapshot_socket/README.mdx:3]: The idea was to use multiples of 10 for the existing gadgets and then start filling the spaces with the new ones. But I suppose this approach of changing the position for all of them works as well. I'm wondering if the script you included in the commit message should be a makefile target.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM from a quick inspection. I'm wondering if the scripts you used to update the files (change the index and the digest) should be part of the makefile.

File: gadgets/README.template (hunk 1 above)
Comment (alban) [reply gadgets/README.template:34]: The flags section are generated by `README-flags.template` and it does in-place update.


`README.template` is only used when the gadget's `README.mdx` does not exist yet.

**Comment by alban** [comment]
> I'm wondering if the scripts you used to update the files (change the index and the digest) should be part of the makefile.

I am adding the script for renumbering in the Makefile.

But I don't think the digest update can be done in the Makefile anymore: the templates are updating README.mdx and not README.md anymore. And artifacthub-pkg.yml is using README.md, which is no longer automatically generated. So the Makefile does not have a way to detect when README.md is modified. In the script, I specified the list of gadgets manually because of this:
```
DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
for i in trace_lsm trace_malloc trace_signal trace_ssl trace_tcpconnect  ; do sed -i "s/^digest:.*$/digest: $DATE/g" gadgets/$i/artifacthub-pkg.yml ; done
```

**Comment by mauriciovasquezbernal** [comment]
There was still a missing piece: #3364

**Pull request merged**